services/packages/packages.proto

```diff
@@ -92,14 +93,40 @@ message UpdateRequest {
 
 message UpdateReply { string debug_output = 1; }
 
-message ListInstalledRequest { PackageSystem package_system = 1; }
+
+message SearchRequest { 
+  PackageSystem package_system = 1; 
+  string name = 2;
+  bool installed = 3;
+  bool latest = 4;
+}
+
+message SearchReply {
+  // key: SearchType value: List of PackageInfo for specific SearchType
+  map<int32, PackageInfoList> package_info_map = 1;
```

Comment:
should we do `map<SearchType, PackageInfoList>` instead?

Reply:
I am just thinking there will be different search type for this SearchReply, we may want installed or latest or both of package info. In the future we can also expand to get all available package NEVRAs. Map is a way distinguish, or we can also explicitly set fields like:
```go
message SearchReply {
  PackageInfo installed_package = 1;
  PackageInfo latest_package = 2;
}
``` 
The below one is easier to process.